app/controllers/activities_controller.rb

```diff
@@ -30,4 +30,38 @@ def poll
       format.js
     end
   end
+
+  private
+
+  def period_start
+    DateTime.parse params[:period_start]
+  end
+
+  def period_end
+    if params[:period_end].empty?
+      DateTime.now
+    else
+      DateTime.parse params[:period_end]
+    end
+  end
+
+  def filtering_params
+    params.slice(:user, :type, :period_start, :period_end)
```

Comment:
Do we need some validation around these params? It seems like any user can pass what ever values in the URL: `activities?user=2&type=&period_start=2023-03-20&period_end=2024-02-01`. If I pass a bogus date, it throws a server error. If I pass any user_id, it seems to still use it for querying. Same with `type`.

Reply:
Since we are not submitting any data and we are only querying activities that are already scoped for the current project is it a big issue? The server error with bogus dates is something that could bother me but still, we won't submit it still so maybe I could do something to just prevent it from throwing a server error